### PR TITLE
user filtring and paging

### PR DESCRIPTION
**Pull request by Eryczeslaw**

### CODE REVIEW

File: HeRoBackEnd/Controllers/UserController.cs [hunk 1]
```diff
@@ -1,25 +1,62 @@
...
         }
 
-        public IActionResult Index()
+        public IActionResult GetList(PagingViewModel pagingViewModel)
         {
-            //List<User> users = usersService.GetAllActive();
-            
-            //return new JsonResult(users);
-            return View();
+            IEnumerable<User> users = userService.GetUsers();
```
Comment (tkarelus) [inline HeRoBackEnd/Controllers/UserController.cs]: All logic should be in the service. Controller should only prepare DTOs and execute service methods

File: HeRoBackEnd/ViewModels/User/PagingViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using Common.Listing;
+
+namespace HeRoBackEnd.ViewModels.User
```
Comment (tkarelus) [inline HeRoBackEnd/ViewModels/User/PagingViewModel.cs]: This class will be used by many modules, not only 'users'

File: Services/Services/UserService.cs [hunk 3]
```diff
@@ -28,9 +25,61 @@ public void UpdateUser(int id, UserDTO dto)
...
+            {
+                users = users.Where(s => s.Email.Contains(userFiltringDTO.Email));
+            }
+            if (!String.IsNullOrEmpty(userFiltringDTO.UserStatus))
+            {
+                users = users.Where(s => s.UserStatus.Contains(userFiltringDTO.UserStatus));
+            }
+            if (!String.IsNullOrEmpty(userFiltringDTO.RoleName))
+            {
+                users = users.Where(s => s.Role.RoleName.Contains(userFiltringDTO.RoleName));
```
Comment (tkarelus) [inline Services/Services/UserService.cs]: RoleName should be equal to the one from filter

File: Services/Services/UserService.cs [hunk 4]
```diff
@@ -28,9 +25,61 @@ public void UpdateUser(int id, UserDTO dto)
...
-            return userRepository.GetAllUsers();
+            IEnumerable<User> users = userRepository.GetAllUsers();
+
+            if (!String.IsNullOrEmpty(userFiltringDTO.Email))
+            {
+                users = users.Where(s => s.Email.Contains(userFiltringDTO.Email));
+            }
+            if (!String.IsNullOrEmpty(userFiltringDTO.UserStatus))
+            {
+                users = users.Where(s => s.UserStatus.Contains(userFiltringDTO.UserStatus));
```
Comment (tkarelus) [inline Services/Services/UserService.cs]: Status should be equal to the one from filter

File: Services/Services/UserService.cs (hunk 3 above)
Comment (Eryczeslaw) [reply Services/Services/UserService.cs]: you mean instead Contains use Equals?

**Review by tkarelus — APPROVED** [review]

**Pull request merged**